test/built-ins/Temporal/PlainDate/prototype/with/copy-properties-not-undefined.js

```diff
@@ -0,0 +1,28 @@
+// Copyright (C) 2022 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-temporal.plaindate.prototype.with
+description: PreparePartialTemporalFields copies only defined properties of source object
+info: |
+  4. For each value _property_ of _fieldNames_, do
+    a. Let _value_ be ? Get(_fields_, _property_).
+    b. If _value_ is not *undefined*, then
+        ...
+        iii. Perform ! CreateDataPropertyOrThrow(_result_, _property_, _value_).
+includes: [temporalHelpers.js]
+features: [Temporal]
+---*/
+
+const plainDate = new Temporal.PlainDate(2006, 1, 24);
+
+TemporalHelpers.assertPlainDate(plainDate.with({ day: 1 }),
+  2006, 1, "M01", 1,
+  "only the properties that are present in the plain object are copied"
+);
+
+const nextMonth = plainDate.add({ months: 1 });
+TemporalHelpers.assertPlainDate(nextMonth.with({ day: nextMonth.daysInMonth }),
```

Comment:
Is there a particular reason to depend on `add` and `daysInMonth` here?

Reply:
Yeah, we probably only need one `with()` test here - they test the same thing anyway.